lab_2_tokenize_by_bpe/main.py

```diff
@@ -81,10 +189,36 @@ def decode(
     :param end_of_word_token: an end-of-word token
     :return: decoded sequence
     """
+    if not isinstance(encoded_text, list) \
+            or not all(isinstance(el, int) for el in encoded_text) \
+            or not isinstance(vocabulary, dict) \
+            or not (isinstance(end_of_word_token, str) or end_of_word_token is None):
+        return None
+
+    encoded_dict = {}
+    for index, num in enumerate(encoded_text):
+        if num not in encoded_dict:
+            encoded_dict[num] = []
+        encoded_dict[num].append(index)
+
+    decoded_text = [''] * len(encoded_text)
```

Comment:
strange... Just append instead of indexing?

Reply:
I thought it will be faster if we refer to huge dictionary only once instead of looking for the token every time it appears again. Indexing allows me to change them all at the same time, therefore, potentially quicker

I guess, I should just simplify